### DISCUSSION TITLE
add `...requestCongif()` methods to the service to make it easier to build things like batch support addons

### DISCUSSION THREAD

**Discussion by toddgeist** [odata2ts/odata2ts#339] [Ideas]
After looking at this more I can see why you wouldn't want to add Batch support to the library. There isn't a good place to plug it into the existing code, without radically shifting the API.  But I do think that there might be a way to make it easier for people to implement their own batching requests functions while still getting all the awesomeness that this library provides.

What if you provided a method like `getPath` but that returned the entire RequestConfig for what ever you were doing?

```
const getRequest = service.people().getConfig()
const createConfig = service.people().createConfig({first_name: "todd"})
```
Now, I can take these and hand them off to the batch requester. I get the type-safe benefits of this library to construct the Request configs, and I can send them to whatever custom request processor I want.

This was inspired by using getPath() to give me the URLs for my own Batch Requester. And that was helpful.  But getting the full Request options would make it even easier.

What do people think?

**Comment by texttechne** [comment]
I intend to refactor the service operations (like query) to have an intermediate step.
Along these lines:
```ts
const request = mainService.people().query();
const response = request.execute();
```

This would then allow for more flexibility regarding the response, e.g. getting the data directly (request.getData() instead of currently query().data) or applying response converters, etc. This would also allow you to retrieve the URL and request config from this request object.

**Reply by toddgeist** [reply]
I thought about this.  But it seemed like a bigger ask.  But if this is the plan, then YAY, I am all for it. 

Thanks

**Reply by texttechne** [reply]
That's now supported by odata2ts. Batch support is on the roadmap now #475
